yt/data_objects/data_containers.py

```diff
@@ -808,13 +815,56 @@ def create_firefly_object(
 
+        if match_any_particle_types is None:
+            # not specified, switching to (temporary) default
+            issue_deprecation_warning(
+                "match_any_particle_types wasn't specified. Its current default "
+                "value (True) will be changed to False in the future. "
+                "Pass an actual value (True or False) to silence this warning. ",
+                since="4.2",
+            )
+            # Note that all of the tests in test_firefly.py have this explicitly
+            # specified (and labeled) to avoid this warning. They will need to be
+            # changed when this behavior changes in the future
+            match_any_particle_types = True
+
         ## initialize a firefly reader instance
         reader = firefly.data_reader.Reader(
             datadir=datadir, clean_datadir=True, **kwargs
         )
 
         ## Ensure at least one field type contains every field requested
-        # error if any requested field is unknown or ambiguous
+        if match_any_particle_types:
+            # Need to keep previous behavior: single string field names that
+            # are ambiguous should bring in any matching ParticleGroups instead
+            # of raising an error
+            # This can be expanded/changed in the future to include field
+            # tuples containing some sort of special "any" ParticleGroup
+            unambiguous_fields_to_include = []
+            unambiguous_fields_units = []
+            for field, field_unit in zip(fields_to_include, fields_units):
+                if isinstance(field, tuple):
+                    # skip tuples, they'll be checked with _determine_fields
+                    unambiguous_fields_to_include.append(field)
+                    unambiguous_fields_units.append(field_unit)
+                    continue
+                _, candidates = self.ds._get_field_info_helper(field)
+                if len(candidates) < 2:
+                    # Field is unambiguous, add in tuple form
+                    # This should be equivalent to _tupleize_field
+                    unambiguous_fields_to_include.append(candidates[0])
+                    unambiguous_fields_units.append(field_unit)
+                else:
+                    # Field has multiple candidates, add all of them instead
+                    # of original field. Note this may bring in aliases and
+                    # equivalent particle fields
+                    for c in candidates:
+                        unambiguous_fields_to_include.append(c)
+                        unambiguous_fields_units.append(field_unit)
+            fields_to_include = unambiguous_fields_to_include
+            fields_units = unambiguous_fields_units
+        # error if any requested field is unknown or (still) ambiguous
+        # This is also sufficient if match_any_particle_types=False
         dd = self.ds.all_data()
         fields_to_include = dd._determine_fields(fields_to_include)
```

Comment:
Finally got back to this.
I think this should fix it. Basically, I kept the old `fields_to_include = dd._determine_fields(fields_to_include)` behavior but added a switch based on your suggested `match_any_particle_types` flag prior to that. If the flag is True, we try to disambiguate any ambiguous single strings by just replacing them with all possible matching candidates and sending the new list to `_determine_fields`. If the flag is False, we just rely on `_determine_fields`.